usr.sbin/inetd/inetd.h

```diff
@@ -0,0 +1,136 @@
+#ifndef _INETD_H
+#define _INETD_H
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <sys/un.h>
+
+#include <arpa/inet.h>
+
+#include <netdb.h>
+
+#include "pathnames.h"
+
+#ifdef IPSEC
+#include <netipsec/ipsec.h>
+#ifndef IPSEC_POLICY_IPSEC	/* no ipsec support on old ipsec */
+#undef IPSEC
+#endif
+#include "ipsec.h"
+#endif
+
+typedef enum service_type {
+	NORM_TYPE = 0, 
+	MUX_TYPE = 1, 
+	MUXPLUS_TYPE = 2, 
+	FAITH_TYPE = 3
+} service_type;
+
+#define ISMUXPLUS(sep)	((sep)->se_type == MUXPLUS_TYPE)
+#define ISMUX(sep)	(((sep)->se_type == MUX_TYPE) || ISMUXPLUS(sep))
+
+#define	TOOMANY		40		/* don't start more than TOOMANY */
+
+#define CONF_ERROR_FMT "%s line %zu: "
+
+/* Log warning/error with 0 or variadic args with line number and file name */
+
+#define ILV(prio, msg, ...) syslog(prio, CONF_ERROR_FMT msg ".", \
+    CONFIG, line_number __VA_OPT__(,) __VA_ARGS__)
+
+#define WRN(msg, ...) ILV(LOG_WARNING, msg __VA_OPT__(,) __VA_ARGS__)
+#define ERR(msg, ...) ILV(LOG_ERR, msg __VA_OPT__(,) __VA_ARGS__)
+
+/* Debug logging */
+#ifdef DEBUG_ENABLE
+#define DPRINTF(fmt, ...) do {\
+	if (debug) {\
+		fprintf(stderr, fmt "\n" __VA_OPT__(,) __VA_ARGS__);\
+	}\
+} while (0)
+#else
+#define DPRINTF(fmt, ...) __nothing
+#endif
+
+#define DPRINTCONF(fmt, ...) DPRINTF(CONF_ERROR_FMT fmt,\
+	CONFIG, line_number __VA_OPT__(,) __VA_ARGS__)
+
+struct	servtab {
+	char	*se_hostaddr;		/* host address to listen on */
+	char	*se_service;		/* name of service */
+	int	se_socktype;		/* type of socket to use */
+	int	se_family;		/* address family */
+	char	*se_proto;		/* protocol used */
+	int	se_sndbuf;		/* sndbuf size */
+	int	se_rcvbuf;		/* rcvbuf size */
+	int	se_rpcprog;		/* rpc program number */
+	int	se_rpcversl;		/* rpc program lowest version */
+	int	se_rpcversh;		/* rpc program highest version */
+#define isrpcservice(sep)	((sep)->se_rpcversl != 0)
+	pid_t	se_wait;		/* single threaded server */
+	short	se_checked;		/* looked at during merge */
+	char	*se_user;		/* user name to run as */
+	char	*se_group;		/* group name to run as */
+	struct	biltin *se_bi;		/* if built-in, description */
+	char	*se_server;		/* server program */
+#define	MAXARGV 64
+	char	*se_argv[MAXARGV+1];	/* program arguments */
+#ifdef IPSEC
+	char	*se_policy;		/* IPsec poilcy string */
+#endif
+	struct accept_filter_arg se_accf; /* accept filter for stream service */
+	int	se_fd;			/* open descriptor */
+	service_type	se_type;		/* type */
+	union {
+		struct	sockaddr se_ctrladdr;
+		struct	sockaddr_in se_ctrladdr_in;
+		struct	sockaddr_in6 se_ctrladdr_in6; /* in6 is used by bind()/getaddrinfo */
+		struct	sockaddr_un se_ctrladdr_un;
+	};			/* bound address */
+	socklen_t	se_ctrladdr_size;
+	int	se_service_max;		/* max # of instances of this service */
+	int	se_count;		/* number of instances of this service started since se_time */
+	int	se_ip_max;  		/* max # of instances of this service per ip per minute */
```

Comment:
should those counts be size_t? Can they be negative? If not then they should probably be size_t

Reply:
I think se_service_max and se_ip_max can be -1 so that any repeat definitions of them in a key-values service definition can be caught, but that isn't important. We can change it to detect if it is already greater than zero and not worry about if someone sets it to zero and then another value. I put in the manual page that not all errors are necessarily detected.

Reply:
Or you can make the invalid/unititialized value size_t too:
#define UNINIT ((size_t)~0)

Reply:
Sounds perfect 👍.